metrics/internal/collectors/ceph-cluster.go

```diff
@@ -115,3 +124,24 @@ func (c *CephClusterCollector) collectMirrorinDaemonCount(cephClusters []*cephv1
 			cephCluster.Namespace)
 	}
 }
+
+func (c *CephClusterCollector) collectLegacyOSDCount(cephClusters []*cephv1.CephCluster, ch chan<- prometheus.Metric) {
+	for _, cephCluster := range cephClusters {
+		cephStorage := cephCluster.Status.CephStorage
+		legacyOSDCount := 0
+		if cephStorage != nil && cephStorage.DeprecatedOSDs != nil {
+			for _, count := range cephStorage.DeprecatedOSDs {
+				legacyOSDCount += len(count)
+			}
```

Comment:
Since DeprecatedOSDs is a `map[<REASON>][]int{}`, we should only check for the `<REASON>` that we care about. This prevents issues in future if Rook adds more than one `<REASON>`.

Reply:
I kinda disagree, even if there are multiple reasons
Rook is classifying them into deprecated osd for the storagecluster. So we might need to alert the user about it

Reply:
We're only interested in alerting about LVM based OSDs for now.
You'd be correct if you were generating a `ocs_deprecated_osds_count` metric. But, you're generating a `ocs_lvm_osd_count` metric.

Reply:
added a reason for the lvm osd counts